src/Inline/Parser/CloseBracketParser.php

```diff
@@ -89,26 +82,29 @@ public function parse(ContextInterface $context, InlineParserContext $inlineCont
             return false;
         }
 
+        $node = $this->createInline($link['url'], $link['title'], $isImage);
+        $openerNode = $opener->getInlineNode();
+        $openerNode->insertBefore($node);
+        while (($inline = $openerNode->getNext()) !== null) {
+            $node->appendChild($inline);
+        }
+        $openerNode->detach();
+
```

Comment:
This could be simplified to:

```php
$node = $this->createInline($link['url'], $link['title'], $isImage);
$opener->getInlineNode()->replaceWith($node);
while (($inline = $node->getNext()) !== null) {
      $node->appendChild($inline);
}
```